sycl/include/sycl/builtins.hpp

```diff
@@ -2073,19 +2076,8 @@ detail::enable_if_t<detail::is_gentype<T>::value, T> bitselect(T a, T b,
 template <typename T>
 detail::enable_if_t<detail::is_sgentype<T>::value, T> select(T a, T b,
                                                              bool c) __NOEXC {
-  return __sycl_std::__invoke_select<T>(a, b, static_cast<int>(c));
-}
-
-// mgentype select (mgentype a, mgentype b, marray<bool, { N }> c)
-template <typename T,
-          typename = std::enable_if_t<detail::is_mgenfloat<T>::value>>
-sycl::marray<detail::marray_element_t<T>, T::size()>
-select(T a, T b, sycl::marray<bool, T::size()> c) __NOEXC {
-  sycl::marray<detail::marray_element_t<T>, T::size()> res;
-  for (int i = 0; i < a.size(); i++) {
-    res[i] = select(a[i], b[i], c[i]);
-  }
-  return res;
+  return __sycl_std::__invoke_select<T>(
+      a, b, static_cast<detail::get_select_opencl_builtin_c_arg_type<T>>(c));
```

Comment:
I'd prefer `get_select_opencl_builtin_c_arg_type` to be inlined to here.

Reply:
Applied.